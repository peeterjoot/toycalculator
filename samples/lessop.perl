#! /usr/bin/perl

my %v1 = (
   'i1' => 'TRUE',
   'i8' => 10,
   'i16' => 1000,
   'i32' => 100000,
   'i64' => 100000000000,
   'f32' => 1.1,
   'f64' => 0.22,
);
my %v2 = (
   'j1' => 'FALSE',
   'j8' => 1,
   'j16' => 100,
   'j32' => 10000,
   'j64' => 10000000000,
   'g32' => 1.0,
   'g64' => 0.21,
);

open my $toy, ">lessop.toy" or die;
open my $etoy, ">expected/lessop.out" or die;

print $toy qq(//THIS IS A GENERATED TEST CASE (./lessop.perl).  DO NOT EDIT\n
INT32 i;
BOOL b;\n);

my @symbols = sort keys %v1;
push( @symbols, sort keys %v2 );

foreach my $v ( @symbols )
{
    my $type = $v;
    $type =~ s/^[ij]/INT/;
    $type =~ s/^[fg]/FLOAT/;

    $type =~ s/INT1$/BOOL/;

    print $toy "$type $v;\n";
}

foreach my $v ( sort keys %v1 )
{
    print $toy "$v = $v1{$v};\n";
}

foreach my $v ( sort keys %v2 )
{
    print $toy "$v = $v2{$v};\n";
}

my $i = 12340000;
#foreach my $v1 ( (qw(i1)) )
foreach my $v1 ( sort keys %v1 )
{
    #foreach my $v2( (qw(j16)) )
    foreach my $v2( sort keys %v2 )
    {
        my $e, $f;
        my $a = $v1{$v1};
        my $b = $v2{$v2};
        $a =~ s/TRUE/1/;
        $b =~ s/TRUE/1/;
        $a =~ s/FALSE/0/;
        $b =~ s/FALSE/0/;

        if ( $a < $b )
        {
            $e = 1;
        }
        else
        {
            $e = 0;
        }

        if ( $b < $a )
        {
            $f = 1;
        }
        else
        {
            $f = 0;
        }

        my $m = sprintf( "i = %d;\nPRINT i;\n", $i );
        print $etoy "$i\n$e\n";
        print $toy "${m}b = $v1 < $v2;\nPRINT b;\n";
        $i++;

        $m = sprintf( "i = %d;\nPRINT i;\n", $i );
        print $etoy "$i\n$f\n";
        print $toy "${m}b = $v2 < $v1;\nPRINT b;\n";
        $i++;
    }
}

close $toy;
close $etoy;

# vim: et ts=4 sw=4
