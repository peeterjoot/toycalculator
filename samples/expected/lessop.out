f32 < g32
0
g32 < f32
1
f32 < g64
0
g64 < f32
1
f32 < j1
0
j1 < f32
1
f32 < j16
1
j16 < f32
0
f32 < j32
1
j32 < f32
0
f32 < j64
1
j64 < f32
0
f32 < j8
0
j8 < f32
1
f32 < l16
0
l16 < f32
1
f32 < l32
0
l32 < f32
1
f32 < l64
0
l64 < f32
1
f32 < l8
0
l8 < f32
1
f32 < m32
0
m32 < f32
1
f32 < m64
0
m64 < f32
1
f64 < g32
1
g32 < f64
0
f64 < g64
0
g64 < f64
1
f64 < j1
0
j1 < f64
1
f64 < j16
1
j16 < f64
0
f64 < j32
1
j32 < f64
0
f64 < j64
1
j64 < f64
0
f64 < j8
1
j8 < f64
0
f64 < l16
0
l16 < f64
1
f64 < l32
0
l32 < f64
1
f64 < l64
0
l64 < f64
1
f64 < l8
0
l8 < f64
1
f64 < m32
0
m32 < f64
1
f64 < m64
0
m64 < f64
1
h32 < g32
1
g32 < h32
0
h32 < g64
1
g64 < h32
0
h32 < j1
1
j1 < h32
0
h32 < j16
1
j16 < h32
0
h32 < j32
1
j32 < h32
0
h32 < j64
1
j64 < h32
0
h32 < j8
1
j8 < h32
0
h32 < l16
0
l16 < h32
1
h32 < l32
0
l32 < h32
1
h32 < l64
0
l64 < h32
1
h32 < l8
1
l8 < h32
0
h32 < m32
1
m32 < h32
0
h32 < m64
1
m64 < h32
0
h64 < g32
1
g32 < h64
0
h64 < g64
1
g64 < h64
0
h64 < j1
1
j1 < h64
0
h64 < j16
1
j16 < h64
0
h64 < j32
1
j32 < h64
0
h64 < j64
1
j64 < h64
0
h64 < j8
1
j8 < h64
0
h64 < l16
0
l16 < h64
1
h64 < l32
0
l32 < h64
1
h64 < l64
0
l64 < h64
1
h64 < l8
0
l8 < h64
1
h64 < m32
0
m32 < h64
1
h64 < m64
1
m64 < h64
0
i1 < g32
0
g32 < i1
0
i1 < g64
0
g64 < i1
1
i1 < j1
0
j1 < i1
1
i1 < j16
1
j16 < i1
0
i1 < j32
1
j32 < i1
0
i1 < j64
1
j64 < i1
0
i1 < j8
0
j8 < i1
0
i1 < l16
0
l16 < i1
1
i1 < l32
0
l32 < i1
1
i1 < l64
0
l64 < i1
1
i1 < l8
0
l8 < i1
1
i1 < m32
0
m32 < i1
1
i1 < m64
0
m64 < i1
1
i16 < g32
0
g32 < i16
1
i16 < g64
0
g64 < i16
1
i16 < j1
0
j1 < i16
1
i16 < j16
0
j16 < i16
1
i16 < j32
1
j32 < i16
0
i16 < j64
1
j64 < i16
0
i16 < j8
0
j8 < i16
1
i16 < l16
0
l16 < i16
1
i16 < l32
0
l32 < i16
1
i16 < l64
0
l64 < i16
1
i16 < l8
0
l8 < i16
1
i16 < m32
0
m32 < i16
1
i16 < m64
0
m64 < i16
1
i32 < g32
0
g32 < i32
1
i32 < g64
0
g64 < i32
1
i32 < j1
0
j1 < i32
1
i32 < j16
0
j16 < i32
1
i32 < j32
0
j32 < i32
1
i32 < j64
1
j64 < i32
0
i32 < j8
0
j8 < i32
1
i32 < l16
0
l16 < i32
1
i32 < l32
0
l32 < i32
1
i32 < l64
0
l64 < i32
1
i32 < l8
0
l8 < i32
1
i32 < m32
0
m32 < i32
1
i32 < m64
0
m64 < i32
1
i64 < g32
0
g32 < i64
1
i64 < g64
0
g64 < i64
1
i64 < j1
0
j1 < i64
1
i64 < j16
0
j16 < i64
1
i64 < j32
0
j32 < i64
1
i64 < j64
0
j64 < i64
1
i64 < j8
0
j8 < i64
1
i64 < l16
0
l16 < i64
1
i64 < l32
0
l32 < i64
1
i64 < l64
0
l64 < i64
1
i64 < l8
0
l8 < i64
1
i64 < m32
0
m32 < i64
1
i64 < m64
0
m64 < i64
1
i8 < g32
0
g32 < i8
1
i8 < g64
0
g64 < i8
1
i8 < j1
0
j1 < i8
1
i8 < j16
1
j16 < i8
0
i8 < j32
1
j32 < i8
0
i8 < j64
1
j64 < i8
0
i8 < j8
0
j8 < i8
1
i8 < l16
0
l16 < i8
1
i8 < l32
0
l32 < i8
1
i8 < l64
0
l64 < i8
1
i8 < l8
0
l8 < i8
1
i8 < m32
0
m32 < i8
1
i8 < m64
0
m64 < i8
1
k16 < g32
1
g32 < k16
0
k16 < g64
1
g64 < k16
0
k16 < j1
1
j1 < k16
0
k16 < j16
1
j16 < k16
0
k16 < j32
1
j32 < k16
0
k16 < j64
1
j64 < k16
0
k16 < j8
1
j8 < k16
0
k16 < l16
1
l16 < k16
0
k16 < l32
0
l32 < k16
1
k16 < l64
0
l64 < k16
1
k16 < l8
1
l8 < k16
0
k16 < m32
1
m32 < k16
0
k16 < m64
1
m64 < k16
0
k32 < g32
1
g32 < k32
0
k32 < g64
1
g64 < k32
0
k32 < j1
1
j1 < k32
0
k32 < j16
1
j16 < k32
0
k32 < j32
1
j32 < k32
0
k32 < j64
1
j64 < k32
0
k32 < j8
1
j8 < k32
0
k32 < l16
1
l16 < k32
0
k32 < l32
1
l32 < k32
0
k32 < l64
0
l64 < k32
1
k32 < l8
1
l8 < k32
0
k32 < m32
1
m32 < k32
0
k32 < m64
1
m64 < k32
0
k64 < g32
1
g32 < k64
0
k64 < g64
1
g64 < k64
0
k64 < j1
1
j1 < k64
0
k64 < j16
1
j16 < k64
0
k64 < j32
1
j32 < k64
0
k64 < j64
1
j64 < k64
0
k64 < j8
1
j8 < k64
0
k64 < l16
1
l16 < k64
0
k64 < l32
1
l32 < k64
0
k64 < l64
1
l64 < k64
0
k64 < l8
1
l8 < k64
0
k64 < m32
1
m32 < k64
0
k64 < m64
1
m64 < k64
0
k8 < g32
1
g32 < k8
0
k8 < g64
1
g64 < k8
0
k8 < j1
1
j1 < k8
0
k8 < j16
1
j16 < k8
0
k8 < j32
1
j32 < k8
0
k8 < j64
1
j64 < k8
0
k8 < j8
1
j8 < k8
0
k8 < l16
0
l16 < k8
1
k8 < l32
0
l32 < k8
1
k8 < l64
0
l64 < k8
1
k8 < l8
1
l8 < k8
0
k8 < m32
1
m32 < k8
0
k8 < m64
1
m64 < k8
0
